contracts/contracts/buyback/Buyback.sol

```diff
@@ -1,74 +1,59 @@
 // SPDX-License-Identifier: agpl-3.0
 pragma solidity ^0.8.0;
 
-import { Governable } from "../governance/Governable.sol";
+import { Strategizable } from "../governance/Strategizable.sol";
 import "../interfaces/chainlink/AggregatorV3Interface.sol";
 import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { SafeMath } from "@openzeppelin/contracts/utils/math/SafeMath.sol";
 import { UniswapV3Router } from "../interfaces/UniswapV3Router.sol";
 
-contract Buyback is Governable {
+contract Buyback is Strategizable {
     using SafeERC20 for IERC20;
     using SafeMath for uint256;
 
     event UniswapUpdated(address _address);
-    event BuybackFailed(bytes data);
 
     // Address of Uniswap
     address public uniswapAddr;
 
-    // Address of OUSD Vault
-    address public immutable vaultAddr;
-
     // Swap from OUSD
     IERC20 immutable ousd;
 
-    // Swap to OGN
-    IERC20 immutable ogn;
+    // Swap to OGV
+    IERC20 immutable ogv;
 
     // USDT for Uniswap path
     IERC20 immutable usdt;
 
     // WETH for Uniswap path
     IERC20 immutable weth9;
 
-    // Oracles
-    address immutable ognEthOracle;
-    address immutable ethUsdOracle;
+    // Address that receives rewards
+    address public immutable rewardsSource;
 
     constructor(
```

Comment:
nit: let's document each of the constructor arguments using natspec format https://docs.soliditylang.org/en/v0.8.15/natspec-format.html
I think it's a good habit for any new smart contract code we write...

Reply:
Done